### ISSUE TITLE
Cannot provide a polymorphic #:type-name struct

### ISSUE THREAD

**Issue by bennn** [racket/typed-racket#304]
The code:
```
#lang typed/racket/base
(struct (A) foo () #:type-name Foo)
(provide (struct-out foo))
```

The error:
```
tr-poly-type-name.rkt:2:31: struct/c: expected a struct identifier
  at: Foo
  in: (struct/c Foo)
```

**Comment by bennn** [comment]
@lexi-lambda

**Comment by takikawa** [comment]
Fixed in cea5091ee6e720ce38811fa947503d63f0476227

**Comment by lexi-lambda** [comment]
Oh, thanks @takikawa! I meant to take some time to look at this a while ago, but I forgot about it. D:
